<div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom"><a href="https://www.mathjax.org">
    <img style="height:1.25em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" />
</a></span> </div>

$\require{color}\require{mhchem}$

<header style="background-color:#e7e2ee; color:#461d7c; line-height:2em; padding-left:1em; padding-right:1em; bottom:0; float:bottom; text-align:right; float:right; font-family:serif; font-size:small;">Problem 4-39 ⋮ Çengel, Y. A., and Boles, M. A. (2001). <i>Thermodynamics: An Engineering Approach.</i> Boston: McGraw-Hill.</header>  


### Given   
Steam at 75 kPa and 8 percent quality is contained in a spring-loaded piston–cylinder device, as shown in Fig. P4–39, with an initial volume of 2 m³. Steam is now heated until its volume is 5 m³ and its pressure is 225 kPa. 

### Find 
1. Heat transfer in
1. Work out  



### Get Organized

Even dysfunctional homes deserve a decent diagram.

![figure, cb8, 4-39](figures/ch04_cen98179_p04039_lg.jpg)

#### This gets easy if you recognize these things:  
1. Closed system ⋮ constant mass  
1. We do have boundary work  
1. We can resolve both states ⋮ $(P_1,x_1),\ (P_2,v_2)$  
1. We can assume $\rm \Delta KE = \Delta PE \approx 0$  
1. Linear P-V relationship

#### Assumptions
1. $\rm \Delta KE = \Delta PE \approx 0$   
1. Volume will increase. If we're wrong, we'll see it when we calculate boundary work.




$\bf \color{blue} \Huge\textit{Attack}$


Apply the First Law, with subscripts 1 and 2 representing initial and final states, respectively:

\begin{array}{ccccc}
  E_{in} &-& E_{out} &=& \Delta E_{sys} \\
  Q_{in} &-& W_{b,out} &=& U_2 - U_1 \\
\end{array}

We know that the spring will force a linear relationship between pressure and volume, _ie_ $P = \kappa V$. A graphical solution is easiest.


![PV diagram](figures/cb8,4-021,PVdiagram.png)

\begin{gather*}
 W_{\rm b,out} = \int^2_1 P\,{\rm d}V = \frac{1}{2} \left( P_1 + P_2 \right)\left( V_2 - V_1 \right) \\[6pt]
 P_1 = 75\,{\rm kPa} \quad
 P_2 = 225\,{\rm kPa} \quad
 V_1 = 2\,{\rm m^3} \quad
 V_2 = 5\,{\rm m^3} 
\end{gather*}




\begin{gather*}
  Q_\textit{in} = m\left( u_2 - u_1 \right) - W_{\rm b,out} \\[6pt]
  m = \frac{V_1}{v_1}  \qquad v_1 = v \left(P_1, x_1 \right) 
  \qquad u_1 = u \left(P_1, x_1 \right) \qquad  u_2 = h \left( P_2, v_2 \right)  \\
\end{gather*}

In [1]:
using Unitful
using Plots

In [2]:
include("src/marcucius.jl");

running marcucius...
package LaTeXStrings...package Unitful...function roundme...function sopretty...function subscripts...macro sopretty...macro rubber...marcucius done!


In [3]:
#___ given____________
V_1 = 2u"m^3"
V_2 = 5u"m^3"
x_1 = 0.08
P_1 = 75u"kPa"
P_2 = 225u"kPa"

v_f_70kPa = 1.0360e-3u"m^3/kg" 
v_f_80kPa = 1.0380e-3u"m^3/kg" 
v_f_75kPa = roundme((v_f_70kPa + v_f_80kPa)/2,10)

v_g_70kPa = 2.365u"m^3/kg" 
v_g_80kPa = 2.087u"m^3/kg" 
v_g_75kPa = (v_g_70kPa + v_g_80kPa)/2

u_f_70kPa = 376.63u"kJ/kg" 
u_f_80kPa = 391.58u"kJ/kg"
u_f_75kPa = (u_f_70kPa + u_f_80kPa)/2

u_g_70kPa = 2494.5u"kJ/kg" 
u_g_80kPa = 2498.8u"kJ/kg" 
u_g_75kPa = (u_g_70kPa + u_g_80kPa)/2

v_f_200kPa = 1.0605e-3u"m^3/kg"
v_f_250kPa = 1.0672e-3u"m^3/kg"
v_f_225kPa = roundme((v_f_200kPa + v_f_250kPa)/2,10)

v_g_200kPa = 0.8857u"m^3/kg"
v_g_250kPa = 0.7187u"m^3/kg"
v_g_225kPa = roundme((v_g_200kPa + v_g_250kPa)/2,10)

u_f_200kPa = 504.49u"kJ/kg" 
u_f_250kPa = 535.10u"kJ/kg" 
u_f_225kPa = roundme((u_f_200kPa + u_f_250kPa)/2,10)

u_g_200kPa = 2529.5u"kJ/kg" 
u_g_250kPa = 2537.2u"kJ/kg" 
u_g_225kPa = roundme((u_g_200kPa + u_g_250kPa)/2,10)

(@sopretty v_f_75kPa  render left fraction) |> display
(@sopretty v_g_75kPa  render left fraction) |> display
(@sopretty u_f_75kPa  render left fraction) |> display
(@sopretty u_g_75kPa  render left fraction) |> display
(@sopretty v_f_225kPa render left fraction) |> display
(@sopretty v_g_225kPa render left fraction) |> display
(@sopretty u_f_225kPa render left fraction) |> display
(@sopretty u_g_225kPa render left fraction) |> display

L"$v_{ \text{f,75kPa}} = \textstyle 0.001037~\frac{\rm  m^{3} }{\rm  kg }$"

L"$v_{ \text{g,75kPa}} = \textstyle 2.226~\frac{\rm  m^{3} }{\rm  kg }$"

L"$u_{ \text{f,75kPa}} = \textstyle 384.105~\frac{\rm  kJ }{\rm  kg }$"

L"$u_{ \text{g,75kPa}} = \textstyle 2496.65~\frac{\rm  kJ }{\rm  kg }$"

L"$v_{ \text{f,225kPa}} = \textstyle 0.00106385~\frac{\rm  m^{3} }{\rm  kg }$"

L"$v_{ \text{g,225kPa}} = \textstyle 0.8022~\frac{\rm  m^{3} }{\rm  kg }$"

L"$u_{ \text{f,225kPa}} = \textstyle 519.795~\frac{\rm  kJ }{\rm  kg }$"

L"$u_{ \text{g,225kPa}} = \textstyle 2533.35~\frac{\rm  kJ }{\rm  kg }$"

In [4]:
W_b_out = ((P_1 + P_2) * (V_2 - V_1) / 2) |> u"kJ"  # trapezoid

v_1 = v_f_75kPa + x_1 * (v_g_75kPa - v_f_75kPa)
v_2 = v_1 * V_2 / V_1

m   = V_1 / v_1 |> roundme
u_1 = u_f_75kPa + x_1 * (u_g_75kPa - u_f_75kPa)

x_2 = (v_2 - v_f_225kPa)/(v_g_225kPa - v_f_225kPa)
u_2 = u_f_225kPa + x_2 * (u_g_225kPa - u_f_225kPa) |> roundme

Q_in = (m * (u_2 - u_1) + W_b_out) |> roundme


(@sopretty W_b_out render left fraction) |> display
(@sopretty v_1     render left fraction) |> display
(@sopretty m       render left fraction) |> display
(@sopretty v_2     render left fraction) |> display
(@sopretty u_1     render left fraction) |> display
(@sopretty u_2     render left fraction) |> display
(@sopretty Q_in    render left fraction) |> display

L"$W_{ \text{b,out}} = \textstyle 450.0~{\rm  kJ }{\rm  }$"

L"$v_{ \text{1}} = \textstyle 0.17903404~\frac{\rm  m^{3} }{\rm  kg }$"

L"$m = \textstyle 11.1711~{\rm  kg }{\rm  }$"

L"$v_{ \text{2}} = \textstyle 0.4475851~\frac{\rm  m^{3} }{\rm  kg }$"

L"$u_{ \text{1}} = \textstyle 553.1086~\frac{\rm  kJ }{\rm  kg }$"

L"$u_{ \text{2}} = \textstyle 1642.07~\frac{\rm  kJ }{\rm  kg }$"

L"$Q_{ \text{in}} = \textstyle 12614.8967~{\rm  kJ }{\rm  }$"

<div class="btn-success">&nbsp;<div class="badge"></div><i class="fa fa-bomb"></i><strong> Boom</strong></div>
<div class="panel-success"><div class="panel-heading"><div class="badge"></div> 

\begin{equation*}
   Q_{in} = 12614.90\,\text{kJ}
\end{equation*}  


</div></div>